### DISCUSSION TITLE
How to show consent form again?

### DISCUSSION THREAD

**Discussion by Tobi-La** [Poing-Studios/Godot-AdMob-Android-iOS#212] [Q&A]
### Godot version

3.5.3

### Plugin version

v2.1.4

### Phone information

Honor 9x

### Issue description

Hey, 
I have a button in my app where I want the user to be able to reconfigure the consent.
However, calling MobileAds.request_user_consent() doesn't do anything.

From https://github.com/Poing-Studios/godot-admob-android/issues/57 I would understand that I need to reset the consent beforehand, but calling 

	MobileAds.reset_consent_state()
	MobileAds.request_user_consent()
	
doesn't do the trick either.

Sorry if I overlook something, but I don't understand how this works.

Thanks for your help!

### Steps to reproduce

1. Configure consent
2. Try to reconfigure consent

**Comment by gumaciel** [comment]
Hi @Tobi-La you don't need to reset the consent and ask again always, depends of your needs, you want to user be able to request again the consent without reloading the app, if yes, you could create a button which calls this code you mentioned

Are you still with this problem? Try to add a Timer with 1 or 2 seconds to see if appear again the consent form

**Comment by kondor-dev** [comment]
Hi, I've noticed that the documentation page (https://developers.google.com/admob/android/privacy?hl=en) has been updated.

The section "Privacy options" (https://developers.google.com/admob/android/privacy?hl=en#privacy_options) describes how to show consent form again. It says that method showPrivacyOptionsForm() should be called.

Besides, on the bottom of the page there is a big red warning about reset() method: "Caution: This method is intended to be used for testing purposes only. You shouldn't call reset() in production code."

In my opinion we cannot use current MobileAds.reset_consent_state() in the production. It should be updated according to the docs.

**Comment by gumaciel** [comment]
> In my opinion we cannot use current MobileAds.reset_consent_state() in the production. It should be updated according to the docs.

This warning is old, so to clarify this we created a conversation on Mobile Ads SDK group:

According to this link, I understand the AdMob team says it can be used in production: https://groups.google.com/g/google-admob-ads-sdk/c/F_QHhhE5Qro/m/JaCULPL4BAAJ

So I don't really know if it can be used in production

**Comment by markvtailor** [comment]
Hi, I'm using 

> Plugin ver. 3.0.2

> Godot ver. 4.2

I want to revoke user's consent and then immediately show the consent dialog again. The problem is, the code:
`MobileAds.reset_consent_state()
MobileAds.request_user_consent()`
Does not present in the current version.
Code from "Consent revocation" from [here](https://poing-studios.github.io/godot-admob-plugin/privacy/user_messaging_tools/gdpr_iab_support/) doesn't do a thing either.

The only working solution I've found is:
`UserMessagingPlatform.consent_information.reset()`
`UserMessagingPlatform.consent_information.update(request, _on_consent_info_updated_success, _on_consent_info_updated_failure)`

But the problem is, it seems like the amount of forms increments at each call. E.g. on the first call there will be one dialog, on second two dialogs, etc. I was unable to find any solution, so any help will be appreciated,

P.S. English is not my first language, so I apologize if something I said is unclear.

**Comment by xdekasx** [comment]
Hi, 

Maybe it's a bit late but I was in the same trouble, `MobileAds.request_user_consent()` didn't do anything.

My solution was as simply as update to 2.1.7 that updates the play-services-ads to version 23.
Here the release: https://github.com/poingstudios/godot-admob-android/releases/tag/v2.1.7

After update it, consent message is shown in my app using `MobileAds.request_user_consent()`, and is reset using `MobileAds.reset_consent_state()`

Be aware that you also have to set up correctly this message from your Admob account and select your app to be working, or at least this was my case.

Hope it helps!
